.github/workflows/check_cla_assigned.yml

```
@@ -0,0 +1,176 @@
+              const username = process.env.USERNAME;
+              console.log(`Checking CLA signature for user: ${username}`);
+
+              const { google } = require('googleapis');
+              const { JWT } = require('google-auth-library');
+              const { OAuth2Client } = require('google-auth-library');
+
+              if (!sheetsCredJson.installed || !sheetsCredJson.installed.client_id || !sheetsCredJson.installed.client_secret || !sheetsCredJson.refresh_token) {
+                console.error("SHEETS_CRED is missing required credential fields");
+                core.warning("SHEETS_CRED is missing required credential fields");
+                core.setOutput('has-signed-cla', 'true');
+                return;
+              }
+
+              const oauth2Client = new OAuth2Client(
+                sheetsCredJson.installed.client_id,
+                sheetsCredJson.installed.client_secret
+              );
+              oauth2Client.setCredentials({
+                refresh_token: sheetsCredJson.refresh_token
+              });
+
+              const sheets = google.sheets({ version: 'v4', auth: oauth2Client });
+              console.log(`Fetching data from spreadsheet ID: ${process.env.SPREADSHEET_ID}`);
+              const response = await sheets.spreadsheets.values.get({
+                spreadsheetId: process.env.SPREADSHEET_ID,
+                range: 'Usernames!A:A',
+              });
+
+              if (!response) {
+                console.log("No response from Google Sheets API");
+                core.setOutput('has-signed-cla', 'false');
+                return;
+              }
+              if (!response.data.values) {
+                console.log("No values found in spreadsheet");
+                core.setOutput('has-signed-cla', 'false');
+                return;
+              }
+
+              const lowercaseUsername = username.toLowerCase();
+              const hasSignedCla = response.data.values.some(row =>
+                row[0] && row[0].toLowerCase() === lowercaseUsername
+              );
+
+              console.log(`User ${username} has ${hasSignedCla ? '' : 'not '}signed the CLA`);
+              core.setOutput('has-signed-cla', hasSignedCla ? 'true' : 'false');
+
+
+            } catch (error) {
+              console.error("Error during CLA verification:", error);
+              core.warning(`CLA check failed: ${error.message}`);
+              // Don't fail the workflow on error, default to allowing the contribution
+              core.setOutput('has-signed-cla', 'true');
+            }
+
+
+      - name: Remove assignee and notify if CLA not signed
+        if: steps.check-cla.outputs.has-signed-cla == 'false'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // Determine if we're handling a PR or an issue
+            const isPullRequest = !!context.payload.pull_request;
+            const isIssue = !!context.payload.issue;
```

```suggestion
            const isPullRequest = Boolean(context.payload.pull_request);
            const isIssue = Boolean(context.payload.issue);
```

I think this is more readable

---

Hmm, done.